src/node_file.cc

```diff
@@ -103,34 +105,87 @@ using v8::Value;
   if (*path == nullptr)                                                     \
     return TYPE_ERROR( #path " must be a string or Buffer");
 
-FSReqWrap* FSReqWrap::New(Environment* env,
-                          Local<Object> req,
-                          const char* syscall,
-                          const char* data,
-                          enum encoding encoding,
-                          Ownership ownership) {
+FD::FD(Environment* env, int fd)
+    : AsyncWrap(env,
+                env->fd_constructor_template()->NewInstance(env->context())
+                    .ToLocalChecked(),
+                AsyncWrap::PROVIDER_FD),
+      fd_(fd) {
+  CHECK_GE(fd, 3);  // Do not allow wrapping of stdio fd
```

Comment:
Why?

Reply:
To keep myself from doing silly things until it's a bit more baked.